### ISSUE TITLE
Keras callbacks and learning rate schedules

### ISSUE THREAD

**Issue by adamelliotfields** [adamelliotfields/lab#40]
* `ModelCheckpoint`
* `BackupAndRestore`
* `TensorBoard`
* `EarlyStopping`
* `ReduceLROnPlateau`
* `TerminateOnNaN`

Also custom callbacks (lifecycle events), lambda callbacks, and custom learning rate schedules like cosine decay.
